src/solver/page/useSolverSolution.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  solveProduction,
+  useHighs,
+} from '@/solver/algorithm/solveProduction';
+import {
+  useFactoryInputsOutputs,
+} from '@/factories/store/factoriesSelectors';
+import {
+  useCurrentSolverId,
+  usePathSolverInstance,
+  useSolverGameId,
+} from '@/solver/store/solverSelectors';
+import { useEffect, useMemo } from 'react';
+import { useStore } from '@/core/zustand';
+import { Path, setByPath } from '@clickbar/dot-diver';
+import { SolverInstance } from '@/solver/store/Solver';
+import { useFormOnChange } from '@/core/form/useFormOnChange';
+import {
+  ISolverSolutionSuggestion,
+  proposeSolverSolutionSuggestions,
+} from '@/solver/page/suggestions/proposeSolverSolutionSuggestions';
+import { isSolutionFound } from '@/solver/algorithm/solve/isSolutionFound';
+import { loglev } from '@/core/logger/log';
+
+const logger = loglev.getLogger('solver:page');
+
+const DEFAULT_FACTORY = {
+  inputs: [],
+  outputs: [
+    {
+      resource: 'Desc_Cement_C',
+      amount: 20,
+    },
+  ],
+};
+export const useSolverSolution = (id: string, mode: 'game' | 'standalone') => {
+  const { highsRef, loading } = useHighs();
+
+  const currentSolverId = useCurrentSolverId();
+  const inputsOutputs = useFactoryInputsOutputs(id);
+  const instance = usePathSolverInstance(id);
+  // This is not the _displayed_ solver ID, but the one that is to be used if no solver ID is provided
+  const solverGameId = useSolverGameId(id);
+
+  useEffect(() => {
+    if (instance) return;
+
+    logger.info('SolverPage: No instance or factory, creating', id);
+
+    useStore.getState().upsertFactorySolver(id, DEFAULT_FACTORY);
+
+    if (mode === 'standalone') {
+      useStore.getState().updateFactory(id, old => {
+        Object.assign(old, DEFAULT_FACTORY);
```

Comment:
could we just go with `old => ({ ...old, ...DEFAULT_FACTORY })`?

Reply:
The spread syntax assigns both old and DEFAULT_FACTORY properties to a new object IIRC. What I wanted was to assign the DEFAULT_FACTORY properties to the old object

Reply:
Oh ok, you're right